lib/chat_transcript_service.rb

```diff
@@ -20,41 +20,55 @@ class ChatTranscriptService
   NO_LINK_ATTR = "noLink=\"true\""
 
   class ChatTranscriptBBCode
-    attr_reader :channel, :multiquote, :chained, :no_link
+    attr_reader :channel, :multiquote, :chained, :no_link, :include_reactions
 
     def initialize(
       channel: nil,
       multiquote: false,
       chained: false,
-      no_link: false
+      no_link: false,
+      include_reactions: false
     )
       @channel = channel
       @multiquote = multiquote
       @chained = chained
       @no_link = no_link
-      @messages = []
+      @include_reactions = include_reactions
+      @message_data = []
     end
 
-    def <<(message)
-      @messages << message
+    def add(message:, reactions: nil)
+      @message_data << { message: message, reactions: reactions }
     end
 
     def render
-      attrs = [quote_attr(@messages.first)]
+      attrs = [quote_attr(@message_data.first[:message])]
       attrs << channel_attr if channel
       attrs << MULTIQUOTE_ATTR if multiquote
       attrs << CHAINED_ATTR if chained
       attrs << NO_LINK_ATTR if no_link
+      attrs << reactions_attr if include_reactions
 
       <<~MARKDOWN
-      [chat #{attrs.join(" ")}]
-      #{@messages.map(&:to_markdown).join("\n\n")}
+      [chat #{attrs.compact.join(" ")}]
+      #{@message_data.map { |msg| msg[:message].to_markdown }.join("\n\n")}
       [/chat]
       MARKDOWN
     end
 
     private
 
+    def reactions_attr
+      reaction_data = @message_data.map do |msg_data|
+        next if msg_data[:reactions].empty?
+        msg_data[:reactions].map do |react|
+          "#{react.emoji}:#{react.usernames}"
+        end
+      end.compact
```

Comment:
```suggestion
      reaction_data = @message_data.reduce([]) do |array, msg_data|
        next if msg_data[:reactions].empty?

        array << msg_data[:reactions].map do |react|
          "#{react.emoji}:#{react.usernames}"
        end
      end
```

Using `Array#reduce` can simplify the code somewhat here.

Reply:
Nice!